tests/tests_selinux_modules_checksum.yml

```diff
@@ -0,0 +1,56 @@
+---
+- name: Test management of SELinux modules
+  hosts: all
+  become: true
+  become_method: sudo
+  become_user: root
+  gather_facts: true
+  tasks:
+    - name: Execute the role and catch errors
+      when: ansible_distribution == "Fedora" or
+        (ansible_distribution_major_version | int > 8 and
+        ansible_distribution in ["CentOS", "RedHat", "Rocky"]) or
+        (ansible_distribution_major_version | int == 8 and
+        ansible_distribution_minor_version | int >= 6 and
+        ansible_distribution in ["CentOS", "RedHat", "Rocky"])
+      vars:
+        selinux_modules:
+          - {path: "linux-system-roles-selinux-test-a.pp"}
+      block:
+        - name: Execute the role
+          include_role:
+            name: linux-system-roles.selinux
+          register: role_result
+        - name: Get current commit_num
+          shell: cat /var/lib/selinux/targeted/active/commit_num
+          register: commit_num_before
+        - name: Execute the role
+          include_role:
+            name: linux-system-roles.selinux
+          register: role_result
+        - name: Get current commit_num
+          shell: cat /var/lib/selinux/targeted/active/commit_num
+          register: commit_num_after
+      rescue:
```

Comment:
I don't think you need this `rescue` block - if one of the tasks in the `block` list fails, it will report an error, and the `always` block tasks will execute.